e2e/support/helpers/e2e-misc-helpers.js

```diff
@@ -99,28 +99,29 @@ export function interceptPromise(method, path) {
  *   cy.visit(`/dashboard/1`);
  * });
  */
-const cypressWaitAllRecursive = (results, currentCommand, commands) => {
-  return currentCommand.then(result => {
```

Comment:
This wasn't working before. You can try to understand why by doing this.

```ts
const command1 = cy.wrap(1);
const command2 = cy.wrap(2);

command1
  .then(result1 => {
    console.log({ result1 });
    return command2.then(result2 => {
      console.log({ result2 });

      return [result1, result2];
    });
  })
  .then(allResults => {
    console.log({ allResults });
  });
```

Surprisingly you'll get `1`, `2`, and `1` in order (I omitted the key for simplicity, but it would help visualize what's being log when you run them)

In contrast, try running this similar code
```ts
cy.wrap(1)
  .then(result1 => {
    console.log({ result1 });
    return cy.wrap(2).then(result2 => {
      console.log({ result2 });

      return [result1, result2];
    });
  })
  .then(allResults => {
    console.log({ allResults });
  });
```

And guess what! You'll get `1`, `2`, and `[1, 2]` as expected. So what's the deal?

Cypress [`.then()`](https://docs.cypress.io/api/commands/then) will yield the result from the previous command. The difference between the 2 snippets are:
1. In the first snippet, all commands are run up front, so only the result from the first command is returned, because `command2` doesn't yield anything. It's already done when you put it in this line `const command2 = cy.wrap(2)`
2. This is in contrast with the 2nd snippet where we run `cy.wrap(2)` _inside_ the body of `.then()`